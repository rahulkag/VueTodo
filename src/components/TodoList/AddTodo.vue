<template>
    <v-form @submit.prevent="addTodo" ref="addItemForm">
        <v-row>
            <v-col cols="8">
                <v-text-field name="addItem" v-model="itemText" :rules="rules" label="Enter item"
                    autocomplete="off"></v-text-field>
            </v-col>
            <v-col cols="4">
                <v-btn type="submit" block rounded="0" size="x-large" prepend-icon="mdi-plus" color="btnPrimary">
                    Add Item
                </v-btn>
            </v-col>
        </v-row>
    </v-form>
</template>
  
<script>
export default {
    name: "AddTodo",
    data() {
        return {
            itemText: '',
            rules: [
                value => {
                    if (value) return true
                    return 'You must enter a item.'
                },
            ],
        }
    },
    methods: {
        addTodo() {
            if (this.itemText) {
                this.$emit("add-todo", this.itemText);
                this.itemText = '';
                this.snackbar = true
                this.$refs.addItemForm.reset();
            }
        }
    }
}
</script>


