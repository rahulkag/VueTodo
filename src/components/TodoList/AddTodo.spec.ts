import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import AddTodo from './AddTodo.vue'

describe('Add Item', () => {
    const wrapper = mount(AddTodo);
    it('renders properly', () => {
        expect(wrapper.text()).toContain('Add Item')
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find('addItem')).toBeTruthy();
        expect(wrapper.find('v-btn')).toBeTruthy();
        expect(wrapper.find('v-btn').attributes('type')).toEqual('submit');
    })
    // it('should save the value on submit', () => {
    //     expect(wrapper.find('v-btn').attributes('type')).toEqual('submit');
    //     wrapper.get('v-text-field').setValue('this is my item');
    //     wrapper.find('v-btn').trigger('click');
    // })
})
