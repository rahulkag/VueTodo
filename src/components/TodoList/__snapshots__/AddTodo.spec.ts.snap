// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add Item > renders properly 1`] = `
"<v-form>
  <v-row>
    <v-col cols="8">
      <v-text-field name="addItem" modelvalue="" rules="value => {
                    if (value) return true
                    return 'You must enter a item.'
                }" label="Enter item" autocomplete="off"></v-text-field>
    </v-col>
    <v-col cols="4">
      <v-btn type="submit" block="" rounded="0" size="x-large" prepend-icon="mdi-plus" color="btnPrimary"> Add Item </v-btn>
    </v-col>
  </v-row>
</v-form>"
`;
